day2/q2.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+func main() {
+	var wg sync.WaitGroup
+	wg.Add(200)                   // add 200 students to wait group
+	rating := make(chan int, 200) // buffered channel for 200 students
+	for i := 0; i < 200; i++ {
+		go func() {
+			defer wg.Done()
+			time.Sleep(time.Duration(rand.Intn(200)) * time.Millisecond) // to generate random response
+			student := rand.Intn(10)                                     // rating between 1 and 10
+			rating <- student
+		}()
```

Comment:
you can separate out this function, so that code will be more readable and reusable.

Reply:
Okay . Willl change